ucsb-cs156-s24/proj-gauchoride-s24-5pm-6
# Lindsey - Automatically toggle rider role after accepting application feature

- Added function to automatically toggle a member's rider role after their application has been accepted 
- Added test as well, passing npm test and stryker
- Deployment link: https://project-lindseywn.dokku-14.cs.ucsb.edu/
- Make sure to use @ucsb.edu emails when using the deployment link. Alternatively, if no way of using ucsb emails, can test locally by temporarily changing SecurityConfig to include @gmail.com emails.
- Currently, you will not be able to see rider applications that are created from a different account. This is due to a bug in the existing code but I've fixed it in https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-6/pull/40
- Steps to test the automatically toggling feature:
1. Using your UCSB email with admin role as true and rider role as false, click Apply to be a Rider and create an application.
2. Go to Rider Applications and approve your own application.
3. Go to Users table and see that rider role is now true.

Before accepting application (lindseywen@ucsb.edu rider role is false)
<img width="1440" alt="Screen Shot 2024-05-21 at 10 05 04 PM" src="https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-6/assets/122336146/b1f709c3-9ac5-457d-9527-60eb6cf63fe9">

Application created (status: pending)
<img width="1440" alt="Screen Shot 2024-05-21 at 10 05 53 PM" src="https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-6/assets/122336146/c492a0f9-98c7-4ba7-8b56-580402c6c067">

Application accepted (status: accepted)
<img width="1440" alt="Screen Shot 2024-05-21 at 10 06 43 PM" src="https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-6/assets/122336146/b4d24463-8f6d-4062-9113-c4aa5bbd4c01">

After application accepted, lindseywen@ucsb.edu rider role is now true
<img width="1440" alt="Screen Shot 2024-05-21 at 10 07 15 PM" src="https://github.com/ucsb-cs156-s24/proj-gauchoride-s24-5pm-6/assets/122336146/eaf6efb9-a319-4bf0-979b-33e2aa1f1a03">

Closes #4
